runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -11963,3 +11986,295 @@ J9::Z::TreeEvaluator::tabortEvaluator(TR::Node * node, TR::CodeGenerator * cg)
+   if (memRef == NULL)
+      {
+      // MemRef can generate BRCL to unresolved data snippet if needed.
+      loadMemRef = generateS390MemoryReference(node, cg);
+      }
+   else
+      {
+      loadMemRef = generateS390MemoryReference(*memRef, 0, cg);
+      }
+
+   if (!isCompressedRef)
+      {
+      TR::TreeEvaluator::checkAndSetMemRefDataSnippetRelocationType(node, cg, loadMemRef);
+      }
+
+   const bool fieldUnresolved = node->getSymbolReference()->isUnresolved();
+   if (comp->getOption(TR_TraceCG))
+      {
+      traceMsg(comp, "SoftwareReadBarrier: symbol is %s. Compr shift %d. RA reg: %s Entry reg %s\n",
+               fieldUnresolved ? "unresolved" : "resolved",
+               TR::Compiler->om.compressedReferenceShift(),
+               raReg->getRegisterName(comp),
+               cg->getEntryPointRealRegister()->getRegisterName(comp));
+      }
+
+   TR::LabelSymbol* startICFLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol* doneLabel = generateLabelSymbol(cg);
+
+   /*
+    * S390HelperCallSnippet clobbers return address register and may clobber entry point register if a trampoline
+    * is used. Make sure they are both free.
+    */
+   TR::Register* dummyRegForRA = cg->allocateRegister();
+   TR::Register* dummyRegForEntry = cg->allocateRegister();
+   dummyRegForRA->setPlaceholderReg();
+   dummyRegForEntry->setPlaceholderReg();
+
+   TR::RegisterDependencyConditions* deps = generateRegisterDependencyConditions(0, 7, cg);
+   deps->addPostCondition(resultReg, TR::RealRegister::AssignAny);
+   deps->addPostCondition(locationReg, TR::RealRegister::AssignAny);
+   deps->addPostCondition(dummyRegForRA, cg->getReturnAddressRegister());
+   deps->addPostCondition(dummyRegForEntry, cg->getEntryPointRegister());
+
+   // Load address of the target field, to be used by GC evacuation helper.
+   generateS390LabelInstruction(cg, TR::InstOpCode::LABEL, node, startICFLabel);
+   startICFLabel->setStartInternalControlFlow();
+
+   int32_t shiftAmount = TR::Compiler->om.compressedReferenceShift();
+   bool shouldShift = (shiftAmount != 0);
+   TR::InstOpCode::Mnemonic loadOpCode = isCompressedRef ? TR::InstOpCode::LLGF: TR::InstOpCode::LG;
+
+   if (fieldUnresolved)
+      {
+      generateRXInstruction(cg, TR::InstOpCode::LAY, node, locationReg, loadMemRef);
+      generateRXInstruction(cg, loadOpCode, node, resultReg, generateS390MemoryReference(locationReg, 0, cg));
+      }
+   else
+      {
+      generateRXInstruction(cg, loadOpCode, node, resultReg, loadMemRef);
+      }
+   if (shouldShift)
+      {
+      generateRSInstruction(cg, TR::InstOpCode::SLLG, node, resultReg, resultReg, shiftAmount);
+      }
+
+   if (loadMemRef->getIndexRegister() != NULL)
+      {
+      deps->addPostCondition(loadMemRef->getIndexRegister(), TR::RealRegister::AssignAny);
+      }
+   if (loadMemRef->getBaseRegister() != NULL)
+      {
+      deps->addPostCondition(loadMemRef->getBaseRegister(), TR::RealRegister::AssignAny);
+      }
```

Comment:
We have an API for these two. We should use `deps->addAssignAnyPostCondOnMemRef(loadMemRef)`